src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * Implementation of the ReceiverTactic
+ */
+#include "ai/hl/stp/tactic/receiver_tactic.h"
+
+#include "ai/hl/stp/action/move_action.h"
+#include "ai/hl/stp/evaluation/calc_best_shot.h"
+#include "geom/util.h"
+#include "shared/constants.h"
+#include "util/logger/init.h"
+
+using namespace AI::Passing;
+using namespace Evaluation;
+
+ReceiverTactic::ReceiverTactic(const Field& field, const Team& friendly_team,
+                               const Team& enemy_team, const AI::Passing::Pass pass,
+                               const Ball& ball, bool loop_forever)
+    : field(field),
+      pass(pass),
+      ball(ball),
+      friendly_team(friendly_team),
+      enemy_team(enemy_team),
+      Tactic(loop_forever)
+{
+}
+
+std::string ReceiverTactic::getName() const
+{
+    return "Receiver Tactic";
+}
+
+void ReceiverTactic::updateParams(const Team& updated_friendly_team,
+                                  const Team& updated_enemy_team,
+                                  const AI::Passing::Pass& updated_pass,
+                                  const Ball& updated_ball)
+{
+    this->friendly_team = updated_friendly_team;
+    this->enemy_team    = updated_enemy_team;
+    this->ball          = updated_ball;
+    this->pass          = updated_pass;
+}
+
+double ReceiverTactic::calculateRobotCost(const Robot& robot, const World& world)
+{
+    // Prefer robots closer to the pass receive position
+    // We normalize with the total field length so that robots that are within the field
+    // have a cost less than 1
+    double cost =
+        (robot.position() - pass.receiverPoint()).len() / world.field().totalLength();
+    return std::clamp<double>(cost, 0, 1);
+}
+
+std::unique_ptr<Intent> ReceiverTactic::calculateNextIntent(
+    intent_coroutine::push_type& yield)
+{
+    MoveAction move_action = MoveAction(MoveAction::ROBOT_CLOSE_TO_DEST_THRESHOLD, true);
+    while (ball.lastUpdateTimestamp() < pass.startTime())
+    {
+        // We want the robot to move to the receiving position for the shot and also
+        // rotate to the correct orientation to face where the pass is coming from
+        yield(move_action.updateStateAndGetNextIntent(*robot, pass.receiverPoint(),
+                                                      pass.receiverOrientation(), 0));
+    }
+
+    // Check if we can shoot on the enemy goal from the receiver position
+    std::optional<std::pair<Point, Angle>> best_shot_opt =
+        calcBestShotOnEnemyGoal(field, friendly_team, enemy_team, *robot);
+
+    // Vector from the ball to the robot
+    Vector robot_to_ball = ball.position() - robot->position();
+
+    // The angle the robot will have to deflect the ball to shoot
+    Angle abs_deflection_angle;
```

Comment:
So they aren't used because `best_shot_opt` must evaluate to `true` to use later parts of the large conditional, in which case the first `if` case is guaranteed to have entered